guides/data_science/machine_learning/classification_algorithms/KNN.md

```diff
@@ -0,0 +1,39 @@
+# K-Nearest Neighbors Algorithm
+
+**KNN Algorithm** is a simple algorithm that is vastly used for both classification and regression analysis. "K" denotes the number of nearest neighbors that is considered when the object's class or value is assigned.
+'K' is always a positive integer, and naturally a uneven number.
+
+# Why KNN
+
+1. Easy to interpret output
+2. Less calculation time
+3. Good predictive Power
+
+## KNN Classification
+
+The object is assigned with the class that is same with majority of nearest neighbors.
+
+## KNN Regression
+
+The object is assigned with the average of the values of k nearest neighbors.
+
+## KNN Algorithm in simple
+
+1. Pick a value for 'K'.
+2. Calculate the distance (might be eucladian distance) of unknown case from all known cases.
+3. Select the K-observations in the training data that are "nearest"to the unknown data point.
+4. Predic the response of the unknown data point using the most popular response value from the K-nearest neighbors.
+
+## Best value of K
+
+The best choice of k depends upon the data; generally, larger values of k reduces effect of the noise on the classification, but make boundaries between classes less distinct. A good k can be selected by various heuristic techniques.
```

Comment:
What is the essence of "boundaries between classes less distinct"? and what are these techniques?
Explain the article in a sense that even a newbie can understand KNN 😄